x-pack/auditbeat/module/system/socket/socket_linux.go

```diff
@@ -339,6 +364,13 @@ func (m *MetricSet) Cleanup() {
 			m.log.Warnf("Failed to remove KProbes on exit: %v", err)
 		}
 	}
+	if m.mountedFS != nil {
+		if err := m.mountedFS.unmount(); err != nil {
```

Comment:
If another application starts using kprobes after Auditbeat mounts this will it unmount? On one hand it's good to clean up something Auditbeat mounts, on the other hand we can't know if anything else is using it after Auditbeat mounts it. Not requesting any change here.

Reply:
Originally I considered mounting it in some temporary location, but I worried it could introduce more problems than it solves.

Tested by adding a kprobe manually while running and the result is:

> 2019-09-06T14:22:02.542Z	ERROR	[socket]	socket/socket_linux.go:369	Failed to umount tracefs at /sys/kernel/tracing: device or resource busy

There is no timeout going on, it's an immediate error.

Reply:
I think that's good then because it means Auditbeat won't break another app that has started using mount.